### ISSUE TITLE
e2e tests flag "--prom-push-gateway" has no effect

### ISSUE THREAD

**Issue by nielsole** [kubernetes/kubernetes#45947]
TLDR: https://github.com/kubernetes/kubernetes/pull/34597 removed a feature but left documentation and the flag intact.

**Kubernetes version** (use `kubectl version`): 

    Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.0", GitCommit:"fff5156092b56e6bd60fff75aad4dc9de6b6ef37", GitTreeState:"clean", BuildDate:"2017-03-28T16:36:33Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
    Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:52:34Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**: irrelevant
- **OS** (e.g. from /etc/os-release): irrelevant
- **Kernel** (e.g. `uname -a`): irrelevant
- **Install tools**: kops
- **Others**:

**What happened**:
I am trying to set up periodic conformance tests on our cluster to make sure it is working as expected.
Prometheus will be used for monitoring. After `make`ing kubernetes I executed a command along the lines of:

    GINKGO_PARALLEL=y KUBERNETES_PROVIDER=local     KUBE_MASTER=local   ./_output/bin/e2e.test --host="" --test.v=true --ginkgo.skip="\[(Slow|Volume|Serial)\]" --ginkgo.focus="\[Conformance\]" --prom-push-gateway=prometheus-push-gateway.default.svc.cluster.local:9091

The push gateway does never receive the metrics, because the feature was removed.

**What you expected to happen**:
The tests send the metrics to the push gateway.

**How to reproduce it** (as minimally and precisely as possible):
Search for the "PrometheusPushGateway" flag in this repo. It is used nowhere.

**Anything else we need to know**:
The feature was removed in https://github.com/kubernetes/kubernetes/pull/34597 but the flags were left in place and also in the documentation. It is frustrating to see that this flag is just a black hole. 
A solution to this would be to either reinstate the feature or remove the flag from docs and flags.

**Comment by kargakis** [comment]
@kubernetes/sig-testing-misc

**Comment by spiffxp** [comment]
Blerg.  @jayunit100 pointed this out back in the PR that remove the functionality, let's follow through and remove the flags.  Easy PR if someone wants it
